### ISSUE TITLE
Adding wait_for to resolve eventual consistency issues

### ISSUE THREAD

**Issue by mwdchang** [DARPA-ASKEM/data-service#302]
TLDR: Can we add `refresh=wait_for` option for new document getting indexed or being updated? Applies to
- tds_model
- tds_model_config
- tds_workflow

We often have this pattern in Terarium:
- Creating X, then almost immediately
- View X 

For example "create workflow, goto newly create workflow page", "create model config, refresh/list model config".
Because indexing happens at intervals, it may be possible that create returns success, but fetch returns nothing as the document hasn't been indexed yet.  For example:
1. T=0, new cycle
2. T=100, index new model-config
3. T=600, fetch model-config, nothing there yet
4. T=1000, model-config indexed
5. T=1100, fetch model-config, get result.

I think that since these indices house manually human created/updated objects, they will not grow at a rate where tweaking indexing options will cause an issue. I think adding a `refresh=wait_for` here makes sense for when objects/documents are being created.

Love to hear some thoughts

References
- https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html

**Comment by tangopapatime** [comment]
@mwdchang interesting idea.  Would `refresh=true` be a little be too aggressive in our use case?  It does appear that this flag might be able to solve some of the concurrency issues.  Were the server side events we were looking into meant to address this problem w/o having to apply any sort of long polling?  I ES flag also wouldn't address two people looking at the same project, changing something and have that change propagate to anyone looking at the same project.

**Comment by mwdchang** [comment]
> @mwdchang interesting idea. Would `refresh=true` be a little be too aggressive in our use case? It does appear that this flag might be able to solve some of the concurrency issues. Were the server side events we were looking into meant to address this problem w/o having to apply any sort of long polling? I ES flag also wouldn't address two people looking at the same project, changing something and have that change propagate to anyone looking at the same project.

I think refresh=wait_for is a little less disruptive but largely probably doesn't make a difference in these specific cases. I would probably not use refresh=true if we have high volume and/or high velocity, we have neither.

The server-send-events would be addressing a different problem of having multiple clients, one of them is in-sync with the datastore while the others are no in-sync.

**Comment by mwdchang** [comment]
@brandomr @toddroper any thoughts here?

**Comment by brandomr** [comment]
Seems straightforward to add `wait_for`. @toddroper let's add this next week
